calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -176,6 +181,708 @@ def test_validate_arguments(self, case, sys_exit_called):
+    @patch('calico_ctl.container.netns', autospec=True)
+    def test_container_add_ip_previously_assigned(
+            self, m_netns, m_get_pool_or_exit, m_client,
+            m_get_container_info_or_exit, m_enforce_root, m_sys_exit):
+        """
+        Assert that system exits when an ip address is already assigned in pool
+        """
+        # Set up mock object
+        m_client.get_endpoint.side_effect = KeyError
+        m_client.assign_address.return_value = []
+
+        # Call method under test
+        container.container_add('container1', '1.1.1.1', 'interface')
+
+        # Assert
+        m_sys_exit.assert_called_once_with(1)
+
+    @patch('calico_ctl.container.enforce_root', autospec=True)
+    @patch('calico_ctl.container.get_container_id', autospec=True)
+    @patch('calico_ctl.container.client', autospec=True)
+    @patch('calico_ctl.container.netns', autospec=True)
+    def test_container_remove(self, m_netns, m_client,  m_get_container_id,
+                              m_enforce_root):
+        # Set up mock objects
+        m_get_container_id.return_value = 666
+
+        ipv4_nets = set()
+        ipv4_nets.add(IPNetwork(IPAddress('1.1.1.1')))
+        ipv6_nets = set()
+        m_endpoint = MagicMock(spec=Endpoint)
+        m_endpoint.ipv4_nets = ipv4_nets
+        m_endpoint.ipv6_nets = ipv6_nets
+        m_endpoint.endpoint_id = 12
+        ippool = IPPool('1.1.1.1/24')
+        m_client.get_endpoint.return_value = m_endpoint
+        m_client.get_ip_pools.return_value = [ippool]
+
+        # Call method under test
+        container.container_remove('container1')
+
+        # Assert
+        m_enforce_root.assert_called_once_with()
+        m_get_container_id.assert_called_once_with('container1')
+        m_client.get_endpoint.assert_called_once_with(
+            hostname=node.hostname,
+            orchestrator_id=utils.ORCHESTRATOR_ID,
+            workload_id=666
+        )
+        self.assertEqual(m_client.unassign_address.call_count, 1)
+        m_netns.remove_endpoint.assert_called_once_with(12)
+        m_client.remove_workload(node.hostname, utils.ORCHESTRATOR_ID, 666)
+
+    @patch('sys.exit', autospec=True)
+    @patch('calico_ctl.container.enforce_root', autospec=True)
+    @patch('calico_ctl.container.get_container_id', autospec=True)
+    @patch('calico_ctl.container.client', autospec=True)
+    @patch('calico_ctl.container.netns', autospec=True)
+    def test_container_remove_no_endpoint(self, m_netns, m_client,
+                                          m_get_container_id, m_enforce_root,
+                                          m_sys_exit):
+        """
+        Test that system exits when the client cannot obtain an endpoint
+        """
+        # Set up mock objects
+        m_client.get_endpoint.side_effect = KeyError
+
+        # Call function under test
+        self.assertRaises(UnboundLocalError, container.container_remove, 'container1')
```

Comment:
I'm not sure what this `UnboundLocalError` is... It doesn't seem to be referenced in the function.  I think this should raise the SystemExit like the other function you added.